dissect/target/plugins/apps/webservers/caddy.py

```diff
@@ -0,0 +1,128 @@
+                    p = match["log_file"].replace(" {", "").split(" ")[-1]
+
+                    if not p.startswith("/"):
+                        print(f"{p} is a relative path")
+                        for root in found_roots:
+                            root_parent = self.target.fs.path(f"{root}").parent
+                            abs_log_path = self.target.fs.path(f"{str(root_parent)}/{p}").parent
+                            print(abs_log_path)
+                            if abs_log_path.exists() and abs_log_path not in log_paths:
+                                log_paths.append(abs_log_path)
+                    else:
+                        print(f"{p} is an absolute path")
+                        if (path := self.target.fs.path(p).parent).exists():
+                            if path not in log_paths:
+                                log_paths.append(path)
+
+        return log_paths
+
+    def check_compatible(self) -> bool:
+        return len(self.log_paths) > 0
+
+    @plugin.export(record=WebserverRecord)
+    def access(self) -> Iterator[WebserverRecord]:
+        """Parses Caddy V1 CRF and Caddy V2 JSON logs.
+
+        Resources:
+            - https://caddyserver.com/docs/caddyfile/directives/log#format-modules
+        """
+        tzinfo = self.target.datetime.tzinfo
+
+        for path in self.log_paths:
+            for line in open_decompress(path, "rt"):
+                line = line.strip()
+                if not line:
+                    continue
+
+                # Parse a JSON log line
+                if line.startswith('{"'):
+                    try:
+                        log = json.loads(line)
+                    except json.decoder.JSONDecodeError:
+                        self.target.log.warning("Could not decode Caddy JSON log line in file %s", path)
+                        continue
+
+                    yield WebserverRecord(
+                        ts=from_unix(log["ts"]).replace(tzinfo=tzinfo),
+                        remote_ip=log["request"]["remote_ip"],
+                        url=f"{log['request']['method']} {log['request']['uri']} {log['request']['proto']}",
+                        status_code=log["status"],
+                        bytes_sent=log["size"],
+                        source=path.resolve(),
+                        _target=self.target,
+                    )
+
+                # Try to parse a CLF log line
+                else:
+                    match = LOG_REGEX.match(line)
+                    if not match:
+                        self.target.log.warning(
+                            "Could not match Caddy webserver log line with regex format for log line '%s'", line
+                        )
+                        continue
+                    log = match.groupdict()
+
+                    yield WebserverRecord(
+                        ts=datetime.strptime(log["ts"], "%d/%b/%Y:%H:%M:%S %z").replace(tzinfo=tzinfo),
```

Comment:
```suggestion
                        ts=datetime.strptime(log["ts"], "%d/%b/%Y:%H:%M:%S %z"),
```
I notice a `%z` in the `strptime`, so that should already parse this log line to the correct timezone.